src/gameboy/CPU.cpp

```diff
@@ -632,7 +660,7 @@ bool CPU::getHalt() {
 void CPU::halt_op() {
     if (IME) {
         halt = true;
-    } else if (memory->read(INTERRUPT_ENABLE & INTERRUPT_FLAG & 0x1F)) {
+    } else if (memory->read(INTERRUPT_ENABLE & INTERRUPT_FLAG & 0x1F) == 0) {
```

Comment:
I dont think this is what we want to do. `INTERRUPT_ENABLE & INTERRUPT_FLAG & 0x1F` is equal to `0xffff & 0xff0f & 0x1f` which is equal to `0xF`. 
So what the code does is checks if the value on address `0xf` is `0`.

I think this is what we want:
`(memory->read(INTERRUPT_ENABLE) & memory->read(INTERRUPT_FLAG) & 0x1f) == 0`